File: polyglot/seis_native/go/seis_native_kernel.go
package seisnative

import "sort"

type NativeRoadmapItem struct {
	Lane     string
	Score    int
	Evidence []string
	NextStep string
}

var Roadmap = []NativeRoadmapItem{
	{Lane: "Apple First", Score: 100, Evidence: []string{"Swift", "SwiftUI", "AppKit"}, NextStep: "Build Apple native surfaces first."},
	{Lane: "Data AI", Score: 88, Evidence: []string{"Python", "evaluation", "memory"}, NextStep: "Keep intelligence workflows measurable."},
	{Lane: "Systems", Score: 84, Evidence: []string{"Rust", "typed contracts"}, NextStep: "Move shared logic into safe modules."},
	{Lane: "Android", Score: 76, Evidence: []string{"Kotlin", "Java"}, NextStep: "Mirror product intent on Android."},
	{Lane: "Windows", Score: 72, Evidence: []string{"CSharp", "DotNet"}, NextStep: "Define Windows product contracts."},
	{Lane: "Infrastructure", Score: 70, Evidence: []string{"Go", "SQL", "Shell"}, NextStep: "Keep operations auditable and reversible."},
}

func OrderedRoadmap() []NativeRoadmapItem {
	items := append([]NativeRoadmapItem(nil), Roadmap...)
	sort.SliceStable(items, func(left, right int) bool {
		return items[left].Score > items[right].Score
	})
	return items
}

func TopLane() NativeRoadmapItem {
	return OrderedRoadmap()[0]
}

File: polyglot/seis_native/python/seis_native_kernel.py
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum
from typing import Iterable


class NativeLane(str, Enum):
    APPLE_FIRST = "Apple First"
    DATA_AI = "Data AI"
    SYSTEMS = "Systems"
    ANDROID = "Android"
    WINDOWS = "Windows"
    INFRASTRUCTURE = "Infrastructure"


@dataclass(frozen=True)
class NativeRoadmapItem:
    lane: NativeLane
    score: int
    evidence: tuple[str, ...]
    next_step: str


ROADMAP: tuple[NativeRoadmapItem, ...] = (
    NativeRoadmapItem(NativeLane.APPLE_FIRST, 100, ("Swift", "SwiftUI", "AppKit"), "Build Apple native surfaces first."),
    NativeRoadmapItem(NativeLane.DATA_AI, 88, ("Python", "evaluation", "memory"), "Keep intelligence workflows measurable."),
    NativeRoadmapItem(NativeLane.SYSTEMS, 84, ("Rust", "typed contracts"), "Move shared logic into safe modules."),
    NativeRoadmapItem(NativeLane.ANDROID, 76, ("Kotlin", "Java"), "Mirror product intent on Android."),
    NativeRoadmapItem(NativeLane.WINDOWS, 72, ("CSharp", "DotNet"), "Define Windows product contracts."),
    NativeRoadmapItem(NativeLane.INFRASTRUCTURE, 70, ("Go", "SQL", "Shell"), "Keep operations auditable and reversible."),
)


def ordered_lanes(items: Iterable[NativeRoadmapItem] = ROADMAP) -> list[NativeRoadmapItem]:
    return sorted(items, key=lambda item: item.score, reverse=True)


def summary() -> str:
    lines = ["SEIS Native Polyglot Roadmap"]
    for item in ordered_lanes():
        lines.append(f"- {item.lane.value}: {item.score} :: {item.next_step}")
    return "\n".join(lines)


if __name__ == "__main__":
    print(summary())
